File: CS/Task26/MyButtonEditPainter.cs
using System.Drawing;
using DevExpress.XtraEditors.Drawing;

namespace Task26
{
    public class MyDateEditPainter: ButtonEditPainter
    {
        Image fNoteIcon;

        public MyDateEditPainter()
            : base()
        {
            try
            {
                fNoteIcon = Image.FromFile("NoteIcon.bmp");
            }
            catch
            {
                fNoteIcon = null;
            }
        }

        public override void Draw(ControlGraphicsInfoArgs info)
        {
            base.Draw(info);
            DrawNoteIcon(info);
        }

        protected virtual void DrawNoteIcon(ControlGraphicsInfoArgs info)
        {
            MyDateEditViewInfo VI = info.ViewInfo as MyDateEditViewInfo;

            if (VI.NoteIconRect != Rectangle.Empty)
                if (fNoteIcon == null)
                {
                    info.Graphics.FillEllipse(Brushes.Red, VI.NoteIconRect);
                }
                else
                    info.Graphics.DrawImage(fNoteIcon, VI.NoteIconRect);
        }
    }
}

File: VB/Task26/MyButtonEditPainter.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports DevExpress.XtraEditors.Drawing

Namespace Task26
	Public Class MyDateEditPainter
		Inherits ButtonEditPainter
		Private fNoteIcon As Image

		Public Sub New()
			MyBase.New()
			Try
				fNoteIcon = Image.FromFile("NoteIcon.bmp")
			Catch
				fNoteIcon = Nothing
			End Try
		End Sub

		Public Overrides Sub Draw(ByVal info As ControlGraphicsInfoArgs)
			MyBase.Draw(info)
			DrawNoteIcon(info)
		End Sub

		Protected Overridable Sub DrawNoteIcon(ByVal info As ControlGraphicsInfoArgs)
			Dim VI As MyDateEditViewInfo = TryCast(info.ViewInfo, MyDateEditViewInfo)

			If VI.NoteIconRect <> Rectangle.Empty Then
				If fNoteIcon Is Nothing Then
					info.Graphics.FillEllipse(Brushes.Red, VI.NoteIconRect)
				Else
					info.Graphics.DrawImage(fNoteIcon, VI.NoteIconRect)
				End If
			End If
		End Sub
	End Class
End Namespace
